### PR TITLE
test: Creation of a test suite

### PR DESCRIPTION
**Pull request by raaijmag**
### Major Update: Complete Test Suite Integration

This pull request introduces a comprehensive test suite, alongside minor improvements in the **AmpliGone** package to enhance static analysis and ease of testing. Below is a summary of the major new features included:  

1. **New GitHub Actions Workflow for Static Analysis and Testing**  
   This workflow automates code quality checks and testing through the following steps:  
   - **Formatting**: Ensures consistent code style.  
   - **Linting**: Identifies and resolves potential issues in the codebase.  
   - **Static Code Analysis**: Analyzes the code for deeper issues without execution.  
   - **Tests**: Executes all relevant test cases to ensure reliability.  

2. **Unit Tests for Core Modules**  
   Detailed unit tests have been implemented for the following modules:  
   - `args.py`  
   - `cut_reads`  
   - `cutlery`  
   - `fasta2bed`  

3. **End-to-End (E2E) Testing Framework**  
   A modular and easily extendable framework to simulate real-world scenarios, currently covering:  
   - **SARS-CoV-2 Happy Flow**: Validates the primary functionality with SARS-CoV-2 data.  
   - **Synthetic Data Happy Flows**: Tests functionality using synthetic `.fasta` and `.bed` primer files.  
   - **Edge Cases**:  
     - Empty input, reference, and primer files.  
     - Corrupted input files.  
     - Reads in incorrect FASTQ format.  
     - Reads with sequences that are too short.

### CODE REVIEW

**Comment by raaijmag** [comment]
There are some e2e tests that fail. While I did not find a solution for them yet, I am quite sure that they are not faulty tests but actual issues.

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+      run: |
+        source ~/.venv/bin/activate
+        pylint --fail-under=9 --disable=C0301 AmpliGone/ tests/ > ${{ env.REPORT_PATH }}pylint-report.txt
+
+    # E501: Line too long, W503: Line break before binary operator, E203: Whitespace before ':'
+    # The last two make it non-PEP8 compliant, but are automatically done by black.
+    - name: Run flake8
+      run: |
+        source ~/.venv/bin/activate
+        flake8 --ignore=E501,W503,E203 AmpliGone/ tests/ --output-file=${{ env.REPORT_PATH }}flake8-report.txt
```
Comment (florianzwagemaker) [inline .github/workflows/tests.yaml:97]: Can we replace these individual linters and formatters above by a single instance? not now but potentially later on for easier workflow maintenance.
i.e. superlinter?

File: .github/workflows/tests.yaml [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    - name: Run mypy
+      run: |
+        source ~/.venv/bin/activate
+        mypy --disallow-untyped-defs --disallow-incomplete-defs --ignore-missing-imports --disallow-untyped-decorators --strict-equality \
+        --warn-redundant-casts --warn-unused-ignores --warn-return-any --warn-unreachable AmpliGone/ tests/ > ${{ env.REPORT_PATH }}mypy-report.txt
+
+    - name: Run bandit
+      run: |
+        source ~/.venv/bin/activate
+        bandit -r AmpliGone/ -f json -o ${{ env.REPORT_PATH }}bandit-report.json
```
Comment (florianzwagemaker) [inline .github/workflows/tests.yaml:124]: Do we require Bandit to run as an individual tool when it's also included in the codefactor reports?

File: AmpliGone/log.py [hunk 3]
```diff
@@ -6,7 +39,7 @@
 # Central logging object using Rich's logging library
 FORMAT = "%(message)s"
 logging.basicConfig(
-    level="INFO",
+    level="DEBUG",
```
Comment (florianzwagemaker) [inline AmpliGone/log.py:42]: Default log level should stay at INFO and not DEBUG. This currently kinda breaks the logging output
